media/server/gstplayer/source/GstTextTrackSink.cpp

```diff
@@ -368,12 +379,57 @@ static void gst_rialto_text_track_sink_set_property(GObject *object, guint propI
 
         break;
     }
+    case PROP_POSITION:
+    {
+        guint64 position = g_value_get_uint64(value);
+        std::unique_lock lock{priv->m_mutex};
+        if (priv->m_textTrackSession && priv->m_capsSet)
+        {
+            priv->m_textTrackSession->setPosition(position / GST_MSECOND);
+        }
+        else
+        {
+            priv->m_queuedPosition = position;
+        }
+        break;
+    }
     default:
         G_OBJECT_WARN_INVALID_PROPERTY_ID(object, propId, pspec);
         break;
     }
 }
 
+static gboolean gst_rialto_text_track_sink_query(GstElement *element, GstQuery *query) // NOLINT(build/function_format)
+{
+    GstRialtoTextTrackSink *sink = GST_RIALTO_TEXT_TRACK_SINK(element);
+    GST_DEBUG_OBJECT(sink, "handling query '%s'", GST_QUERY_TYPE_NAME(query));
+    switch (GST_QUERY_TYPE(query))
+    {
+    case GST_QUERY_POSITION:
+    {
+        GstFormat fmt;
+        gst_query_parse_position(query, &fmt, NULL);
+        switch (fmt)
+        {
+        case GST_FORMAT_TIME:
+        {
+            // Thunder ITextTrack does not provide getPosition API so we are unable to determine current position
```

Comment:
In case of position query, we shouldn't return TextTrack's position even if it would supply such option.
That would confuse gstreamer's pipeline.

Reply:
comment modified